pkg/agent/endpoints/workload/handler.go

```diff
@@ -375,24 +382,28 @@ func composeX509SVIDResponse(update *cache.WorkloadUpdate) (*workload.X509SVIDRe
 	return resp, nil
 }
 
-func sendJWTBundlesResponse(update *cache.WorkloadUpdate, stream workload.SpiffeWorkloadAPI_FetchJWTBundlesServer, log logrus.FieldLogger, allowUnauthenticatedVerifiers bool) (err error) {
+func sendJWTBundlesResponse(update *cache.WorkloadUpdate, stream workload.SpiffeWorkloadAPI_FetchJWTBundlesServer, log logrus.FieldLogger, allowUnauthenticatedVerifiers bool, previousResponse *workload.JWTBundlesResponse) (*workload.JWTBundlesResponse, error) {
 	if !allowUnauthenticatedVerifiers && !update.HasIdentity() {
 		log.WithField(telemetry.Registered, false).Error("No identity issued")
-		return status.Error(codes.PermissionDenied, "no identity issued")
+		return nil, status.Error(codes.PermissionDenied, "no identity issued")
 	}
 
 	resp, err := composeJWTBundlesResponse(update)
 	if err != nil {
 		log.WithError(err).Error("Could not serialize JWT bundle response")
-		return status.Errorf(codes.Unavailable, "could not serialize response: %v", err)
+		return nil, status.Errorf(codes.Unavailable, "could not serialize response: %v", err)
+	}
+
+	if reflect.DeepEqual(resp.GetBundles(), previousResponse.GetBundles()) {
```

Comment:
DeepEqual won't do the right thing here. We should use `proto.Equal` instead. We should also compare the full response instead of just the bundle fields (more robust in the face of additional fields). 
```suggestion
	if proto.Equal(resp, previousResponse) {
```